gemrb/core/Scriptable/Actor.cpp

```diff
@@ -2780,6 +2780,20 @@ void Actor::RefreshEffects(bool first, const stats_t& previous)
 		if (!(BaseStats[IE_STATE_ID] & STATE_DEAD)) pcf_hitpoint(this, 0, BaseStats[IE_HITPOINTS]);
 	}
 
+	// apply haste bonuses to attacks per round, possibly exceeding the limit of 5
+	// this must be done after ALL other modifiers, including skill effects in RefreshPCStats() above
+	if (GetStat(IE_STATE_ID) & STATE_HASTED) {
+		int apr = GetStat(IE_NUMBEROFATTACKS);
+		if (GetStat(IE_ATTACKNUMBERDOUBLE)) { // weak haste
+			if (apr % 2) apr++; // round up
+		} else if (GetStat(IE_IMPROVEDHASTE)) {
+			apr += apr; // x2
+		} else { // normal haste
+			apr += 2 - apr % 2; // +1, round down
+		}
+		Modified[IE_NUMBEROFATTACKS] = apr; // do NOT call SetStat or it will be clamped
```

Comment:
Working on Modified here is fine, but the comment shouldn't be needed. Just increase the limit in the maximum_* array.

Reply:
Then _anything_ will be able to exceed 5 APR, which is not the original game behavior.  E.g. level 13 fighter with two stars dual-wielding Belm and Kundane would have 1 + 0.5 + 2 + 1 + 1 = 5.5 APR, but the original game clamps them to 5.  But haste, and haste alone, can exceed that to 6 or 10.  I've commented above that wspatck bonuses are properly clamped by gemrb, which is correct behavior.  But haste is, and must be, an exception.

Reply:
Ok, then I suggest you just remove the comment.

Reply:
I can do that, if you don't think it's helpful.  No big deal.

Reply:
To avoid confusion, yeah.